command/meta.go

```diff
@@ -76,3 +76,44 @@ func (m *meta) client() *api.Client {
 		Port: *m.port,
 	}, nil)
 }
+
+// requestUserApproval returns an exit code and boolean describing if the user
+// approved. If the user did not approve (false is returned), exit code is provided.
+func (m *meta) requestUserApproval(taskName string) (int, bool) {
+	m.UI.Info("Enabling the task will perform the actions described above.")
+	m.UI.Output(fmt.Sprintf("Do you want to perform these actions for '%s'?", taskName))
+	m.UI.Output(" - This action cannot be undone.")
+	m.UI.Output(" - CTS cannot guarantee Terraform will perform these exact actions if")
+	m.UI.Output("   monitored services have changed.\n")
+	m.UI.Output("Only 'yes' will be accepted to approve.\n")
+	v, err := m.UI.Ask(fmt.Sprintf("Enter a value:"))
+	m.UI.Output("")
+
+	if err != nil {
+		m.UI.Error(fmt.Sprintf("Error asking for approval: %s", err))
+		return ExitCodeError, false
+	}
+	if v != "yes" {
+		m.UI.Output(fmt.Sprintf("Cancelled enabling task '%s'", taskName))
+		return ExitCodeOK, false
+	}
+
+	return 0, true
+}
+
+// changesDetected attempts to detect if a change plan has been generated.
+// Does a few string parse checks to try to prevent a false negative
+func (m *meta) changesDetected(plan string) bool {
+	plan = strings.TrimSpace(plan)
+
+	// look for phrases that there are changes
+	if strings.HasPrefix(plan, "An execution plan has been generated") {
```

Comment:
Looking for these phrases is a little iffy. I have in my to-dos to write an e2e test to check terraform output for these phrases. If people have other ideas, let me know!

Reply:
I'm not sure if this would be accessible from this package, but it looks like the tf-exec package has a boolean to provide info on diffs

> The returned boolean is false when the plan diff is empty (no changes) and true when the plan diff is non-empty (changes present).

https://pkg.go.dev/github.com/hashicorp/terraform-exec@v0.13.0/tfexec#Terraform.Plan

Reply:
Ooo thanks for finding this. This seems way better than trying to parse for changes. Let me look into consuming the 'changes present' boolean.

Reply:
The work to consume the 'changes present' boolean got a bit large so I opened a separate PR: https://github.com/hashicorp/consul-terraform-sync/pull/200. I'll rebase this PR off of those changes once they are merged.

Reply:
Rebased! Thanks for this finding @findkim !